src/Hill.jl

```diff
@@ -36,32 +36,38 @@ end
 
 
 """ Hill optimization function. """
-function optimize_hill(conc::Vector, g1::Matrix, g2::Matrix; maxstep = 100000)
+function optimize_hill(conc::Vector, g1::Matrix, g2::Matrix; maxstep = 300000)
+
     f(x) = residHill(x, conc, g1, g2)
 
-    low = [minimum(conc), 1e-9, 1e-9, 0.1, 1e-9, 1e-9, 0.0, 0.0, 0.25, 3, 5, 0, 0]
-    high = [maximum(conc), 1.0, 1.0, 10.0, 1.0, 1.0, 3.0, 3.0, 0.75, 50, 50, 50, 50]
+    # [EC50, k, min_a1, max_a1, min_a2, max_a2, min_b1, max_b1, min_b2, max_b2, max_g11, max_g12, max_g21, max_g22, min%G1]
+    low = [minimum(conc), 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 1e-9, 0.25]
+    high = [maximum(conc), 10.0, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 0.75]
 
     return optimize_helper(f, low, high, maxstep)
 end
 
-""" A function to convert the estimated hill parameters back to ODE parameters. """
-function getODEparams(p::Vector, concentrations::Vector{Float64})
-    effects = Matrix{eltype(p)}(undef, 9, length(concentrations))
-
+function getODEparams(p, conc, nMax)
+    effects = zeros(eltype(p), 9, length(conc[:, 1]), nMax)
```

Comment:
Can't nMax be derived from the length of `p`? If so, this would help prevent errors by not duplicating information.

Reply:
fixed.